resin/settings.py

```diff
@@ -35,15 +36,20 @@ class Settings(object):
                                             'cache')
 
     def __init__(self):
-        if Path.isdir(self._setting['data_directory']):
+        try:
+            self.__read_settings()
+        except:
+            # move old settings file if it exists.
             if Path.isfile(Path.join(self._setting['data_directory'],
                                      self.CONFIG_FILENAME)):
-                self.__read_settings()
-            else:
-                self.__write_settings()
-        else:
-            # data directory doesn't exist
+                shutil.move(Path.join(self._setting['data_directory'],
```

Comment:
There's a race condition here. By the time ``shutil.move`` executes, the original config file may have been moved and your code will crash. When dealing with the filesystem it's best to try to do something and handle any errors. I'd rather you wrote it like so:

```
try:
  shutil.move(config_path, '{0}.{1}'.format(config_path, 'old'))
except OSError:
  pass
```

You should then remove the check for the file's existence in the ``if`` above.